orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/OrcaMessageHandler.kt

```diff
@@ -116,8 +116,21 @@ internal interface OrcaMessageHandler<M : Message> : MessageHandler<M> {
   fun PipelineExecution.shouldQueue(): Boolean {
     val configId = pipelineConfigId
     return when {
-      !isLimitConcurrent -> false
       configId == null -> false
+      !isLimitConcurrent -> {
+        return when {
+          getMaxConcurrentExecutions() <= 0 -> false
+          else -> {
+            val criteria = ExecutionCriteria().setPageSize(maxConcurrentExecutions+2).setStatuses(RUNNING)
```

Comment:
What's this magic 2?

Reply:
I added 2, because of the code that checks for any **RUNNING** pipeline a few lines below starts with a page size of 2. Plus 1 may work, but to be in the safe side I added 2:
`val criteria = ExecutionCriteria().setPageSize(2).setStatuses(RUNNING)`

Reply:
Makes sense, might make sense to extract it out to a constant so we don't end up peppering numbers throughout

Reply:
I'll change the `2` into a constant set to 2. I can name it 'MIN_PAGE_SIZE`.